### ISSUE TITLE
Allow users to select custom countries, cause areas, organizations and etc, using a stackoverflow-like tag selector

### ISSUE THREAD

**Issue by victor-yunenko** [rtcharity/eahub.org#1003]
This is certainly a substantial task that should be considered carefully, since it would drain a significant amount of effort from other potentially more impactful improvements.

![image](https://user-images.githubusercontent.com/36416236/98847705-544db480-2459-11eb-82df-e7d9b6246d24.png)

The technical specification is relatively straightforward though:
- django `GenericForeignKey` Tag model with `name` and `description` fields
- django rest api + vuejs frontend with algolia index search
- a notification for the admins that would allow them to drop or merge the tags created by users (duplicates or simply undesirable tags)

**Comment by victor-yunenko** [comment]
There's also the go-to package for django tags - django-taggit, but its interface is primary suitable for admin panel as far as I can see. Yet we might consider it as a deadly cheap alternative.

**Comment by victor-yunenko** [comment]
For frontend we could use eg http://www.vue-tags-input.com/#/

**Comment by victor-yunenko** [comment]
My guesstimate would be 20-50h.

**Comment by vagarwalla** [comment]
- Thanks for the time estimate! I think that this task is probably worth it but we can delay to Q1/Q2 of 2021.
- Recommend not having "custom countries" since we have a country tag already
- Suggestions for the kinds of tags people should add in grey text. E.g. _Use these tags to indicate important organisational, identity, or other affiliations such as your university, school, uncommon cause areas etc._

**Comment by victor-yunenko** [comment]
I agree about dropping `custom countries` - I think the only use case it had was the ability to select several countries at once, which should be done with a multi-select in any case 👌 

Also I recommend to treat that ticket as the first milestone for any field, and discuss what specific fields should be based on it in different issues.

This implementation ideally would be a django package not related to eahub, since we're hardly the only project who could use it.
